lib/Firtool/Firtool.cpp

```diff
@@ -116,6 +116,8 @@ LogicalResult firtool::populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
   pm.addNestedPass<firrtl::CircuitOp>(firrtl::createLowerFIRRTLTypesPass(
       opt.getPreserveAggregate(), firrtl::PreserveAggregate::None));
 
+  pm.addNestedPass<firrtl::CircuitOp>(firrtl::createSpecializeLayersPass());
```

Comment:
Just wondering -- why at this point in the pipeline? Would it make sense to run this earlier? Motivation for this question is I'm wondering if we're deleting things doing it earlier is less work for everything after that, as one consideration anyway.

Disabling layers (and thereby dropping ports, modules, so on) might mask errors that would otherwise occur if not specializing, so dunno.

Reply:
We talked about this offline: we decided that it would be good to move this earlier in the pipeline but it must run after all the diagnostic passes have run.  We ended up moving it later in the pipeline, after expand whens and check comb loops.  There is some work to create a diagnostic only version of expand whens that is diagnostic only and runs very early in the pipeline and we can maybe move this pass earlier once that is merged.